modules/bluefish/util/blue_velvet.cpp

```diff
@@ -20,67 +20,205 @@
+			GET_PROCADDR_FOR_FUNC(bfcUtilsGetStringForVideoMode, h_module_.get());
+			GET_PROCADDR_FOR_FUNC(bfcUtilsGetStringForMemoryFormat, h_module_.get());
+			GET_PROCADDR_FOR_FUNC(bfcUtilsGetMR2Routing, h_module_.get());
+			GET_PROCADDR_FOR_FUNC(bfcUtilsSetMR2Routing, h_module_.get());
+			GET_PROCADDR_FOR_FUNC(bfcUtilsGetAudioOutputRouting, h_module_.get());
+			GET_PROCADDR_FOR_FUNC(bfcUtilsSetAudioOutputRouting, h_module_.get());
+			GET_PROCADDR_FOR_FUNC(bfcUtilsIsVideoModeProgressive, h_module_.get());
+			GET_PROCADDR_FOR_FUNC(bfcUtilsIsVideoMode1001Framerate, h_module_.get());
+			GET_PROCADDR_FOR_FUNC(bfcUtilsGetFpsForVideoMode, h_module_.get());
+			GET_PROCADDR_FOR_FUNC(bfcUtilsGetVideoModeForFrameInfo, h_module_.get());
+			GET_PROCADDR_FOR_FUNC(bfcUtilsGetFrameInfoForVideoMode, h_module_.get());
+			GET_PROCADDR_FOR_FUNC(bfcUtilsGetAudioSamplesPerFrame, h_module_.get());
+			res = true;
+		}
+		return res;
+	}
 
-	auto module = LoadLibrary(u16(module_str).c_str());
-	if(!module)
-		LoadLibrary(u16(std::string(getenv("SystemDrive")) + "\\Program Files\\Bluefish444\\Driver\\" + module_str).c_str());
-	if(!module)
-		LoadLibrary(u16(std::string(getenv("SystemDrive")) + "\\Program Files (x86)\\BlueFish444\\Driver\\" + module_str).c_str());
-	if(!module)
-		CASPAR_THROW_EXCEPTION(file_not_found() << msg_info("Could not find BlueVelvet3.dll. Required drivers are not installed."));
-	static std::shared_ptr<void> lib(module, FreeLibrary);
-	BlueVelvetFactory4 = reinterpret_cast<decltype(BlueVelvetFactory4)>(GetProcAddress(module, "BlueVelvetFactory4"));
-	BlueVelvetDestroy  = reinterpret_cast<decltype(BlueVelvetDestroy)>(GetProcAddress(module, "BlueVelvetDestroy"));
-	BlueVelvetVersion  = reinterpret_cast<decltype(BlueVelvetVersion)>(GetProcAddress(module, "BlueVelvetVersion"));
-}
+	const char * bvc_wrapper::get_version()
+	{
+		return bfcGetVersion();
+	}
 
-void blue_hanc_initialize()
-{
-#ifdef _DEBUG
-	std::string module_str = "BlueHancUtils64_d.dll";
-#else
-	std::string module_str = "BlueHancUtils64.dll";
-#endif
-	
-	auto module = LoadLibrary(u16(module_str).c_str());
-	if(!module)
-		LoadLibrary(u16(std::string(getenv("SystemDrive")) + "\\Program Files\\Bluefish444\\Driver\\" + module_str).c_str());
-	if(!module)
-		LoadLibrary(u16(std::string(getenv("SystemDrive")) + "\\Program Files (x86)\\BlueFish444\\Driver\\" + module_str).c_str());
-	if(!module)
-		CASPAR_THROW_EXCEPTION(file_not_found() << msg_info("Could not find BlueHancUtils.dll. Required drivers are not installed."));
-	static std::shared_ptr<void> lib(module, FreeLibrary);
-	encode_hanc_frame	 = reinterpret_cast<decltype(encode_hanc_frame)>(GetProcAddress(module, "encode_hanc_frame"));
-	encode_hanc_frame_ex = reinterpret_cast<decltype(encode_hanc_frame_ex)>(GetProcAddress(module, "encode_hanc_frame_ex"));
-}
+	BLUE_UINT32 bvc_wrapper::attach(int iDeviceId)
+	{
+		return bfcAttach((BLUEVELVETC_HANDLE)bvc_.get(), iDeviceId);
```

Comment:
None of these casts to `BLUEVELVETC_HANDLE` should be needed since `bvc_.get()` returns `void*` anyway.